testsupport/tiers/checks.go

```diff
@@ -48,333 +47,234 @@ var (
+			checks = append(checks, commonNetworkPolicyChecks...)
+			checks = append(checks, networkPolicyAllowFromCRW(), numberOfNetworkPolicies(6))
+			return checks
+		default:
+			checks := append(commonChecks,
+				limitRange(defaultCPULimit, "750Mi", "10m", "64Mi"),
+				rbacEditRoleBinding(),
+				rbacEditRole(),
+				crtadminPodsRoleBinding(),
+				crtadminViewRoleBinding(),
+				execPodsRole(),
+				numberOfToolchainRoles(2),
+				numberOfToolchainRoleBindings(4))
+			checks = append(checks, commonNetworkPolicyChecks...)
+			var otherNamespaceKind string
+			switch nsType {
+			case "dev":
+				otherNamespaceKind = "stage"
+			case "stage":
+				otherNamespaceKind = "dev"
+			}
+			checks = append(checks, networkPolicyAllowFromCRW(), networkPolicyAllowFromOtherNamespace(otherNamespaceKind), numberOfNetworkPolicies(7))
+			return checks
+		}
 	}
-
-	checks = append(checks, append(commonNetworkPolicyChecks(), networkPolicyAllowFromCRW(), numberOfNetworkPolicies(6))...)
-	return checks
 }
 
-func (a *appstudioTierChecks) GetExpectedTemplateRefs(hostAwait *wait.HostAwaitility) TemplateRefs {
-	templateRefs := GetTemplateRefs(hostAwait, a.tierName)
-	verifyNsTypes(hostAwait.T, a.tierName, templateRefs, "appstudio")
-	return templateRefs
+func getExpectedTemplateRefs(tierName string) func(hostAwait *wait.HostAwaitility) TemplateRefs {
+	return func(hostAwait *wait.HostAwaitility) TemplateRefs {
+		switch tierName {
+		case appstudio:
+			templateRefs := GetTemplateRefs(hostAwait, tierName)
+			verifyNsTypes(hostAwait.T, tierName, templateRefs, "appstudio")
+			return templateRefs
+		case base1ns:
+			templateRefs := GetTemplateRefs(hostAwait, tierName)
+			verifyNsTypes(hostAwait.T, tierName, templateRefs, "dev")
+			return templateRefs
+		default:
+			templateRefs := GetTemplateRefs(hostAwait, tierName)
+			verifyNsTypes(hostAwait.T, tierName, templateRefs, "dev", "stage")
+			return templateRefs
+		}
+	}
 }
 
-func (a *appstudioTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
-	return clusterObjectsChecks(
-		clusterResourceQuotaCompute(cpuLimit, "1750m", "7Gi", "15Gi"),
-		clusterResourceQuotaDeployments(),
-		clusterResourceQuotaReplicas(),
-		clusterResourceQuotaRoutes(),
-		clusterResourceQuotaJobs(),
-		clusterResourceQuotaServices(),
-		clusterResourceQuotaBuildConfig(),
-		clusterResourceQuotaSecrets(),
-		clusterResourceQuotaConfigMap(),
-		numberOfClusterResourceQuotas(),
-		idlers(43200, ""))
+func getTierObjectChecks(tierName string) func() []tierObjectCheck {
+	return func() []tierObjectCheck {
+		switch tierName {
+		case advanced, basedeactivationdisabled:
+			return []tierObjectCheck{nsTemplateTier(tierName, 0)}
+		case baselarge:
+			return []tierObjectCheck{nsTemplateTier(tierName, 90)}
+		case baseextended:
+			return []tierObjectCheck{nsTemplateTier(tierName, 180)}
+		case hackathon:
+			return []tierObjectCheck{nsTemplateTier(tierName, 80)}
+		case testtier:
+			return []tierObjectCheck{}
+		default:
```

Comment:
Same comment about default here.

Reply:
refactored here, but not returning an error, because we may want to change the deactivation timeout for custom tiers